cmd/aterm/appdialogs/first_run_prompt.go

```diff
@@ -0,0 +1,188 @@
+// description of what is needed
+func FirstRun(primaryConfigFile, pathToCommonConfig string) (config.TermRecorderConfigOverrides, error) {
+	fmt.Printf("Hi and welcome to the ASHIRT Terminal Recorder.\n\n")
+	fmt.Printf("I couldn't read a configuration file. (I looked here: %v). "+
+		"I think this might be the first run of this application. "+
+		"Before we begin recording, we need to configure this application.\n\n",
+		primaryConfigFile)
+
+	var configData config.TermRecorderConfigOverrides
+
+	// try to read common config
+	content, err := ioutil.ReadFile(pathToCommonConfig)
+	if err == nil {
+		err = json.Unmarshal(content, &configData)
+		if err == nil {
+			fmt.Printf("I was able to load defaults from another ASHIRT application. Let's double check these values.\n\n")
+		}
+	}
+
+	fmt.Println("If the value in [brackets] looks good, simply press enter to accept that value.")
+	configData.APIURL = askForAPIURL(configData.APIURL)
+	configData.AccessKey, configData.SecretKey = askForAccessKeyAndSecret(configData.AccessKey, configData.SecretKey)
+	configData.OutputDir = askForSavePath(configData.OutputDir)
+
+	checkConnection := true
+
+	for checkConnection {
+		fmt.Printf(fancy.ClearLine("Let's check the network connection\n"))
+
+		network.SetBaseURL(*configData.APIURL)
+		network.SetAccessKey(*configData.AccessKey)
+		network.SetSecretKey(*configData.SecretKey)
+
+		var testErr error
+		dialog.DoBackgroundLoading(dialog.SyncedFunc(func() {
+			_, testErr = network.TestConnection()
+		}))
+		if testErr == nil {
+			fmt.Printf("These configurations work.\n")
+			checkConnection = false
+		} else if errors.Is(testErr, network.ErrorConnectionNotFound) {
+			fmt.Printf("It looks like the server is not up or the address is wrong.\n")
+			fix, err := dialog.YesNoPrompt("Do you want to try to fix this?", "", medium)
+			if fix && err == nil {
+				configData.APIURL = askForAPIURL(configData.APIURL)
+			} else {
+				checkConnection = false
+			}
+		} else if errors.Is(testErr, network.ErrorConnectionUnauthorized) {
+			fmt.Printf("The server did not accept your access and secret key.\n")
+			fix, err := dialog.YesNoPrompt("Do you want to try to fix this?", "", medium)
+			if fix && err == nil {
+				configData.AccessKey, configData.SecretKey = askForAccessKeyAndSecret(configData.AccessKey, configData.SecretKey)
+			} else {
+				checkConnection = false
+			}
+		} else {
+			fmt.Printf("I got an error I wasn't expecting. It's: '%v'. "+
+				"This may be due to a network issue with the ASHIRT servers or with your own connection. "+
+				"Please try contacting an administrator for help.\n", testErr.Error())
+			checkConnection = false
+		}
+	}
+
+	fmt.Printf("\nOkay, that should be enough data for now. "+
+		"I will create a configuration file here: %v. "+
+		"You can find additional configuration options there.\n\n", config.ATermConfigPath())
+
+	return configData, nil
+}
+
+func askForAPIURL(guessValue *string) *string {
```

Comment:
When updating the api url from the update settings menu there is a space before the prompt. Can we update the copy to read "I need to know how to reach the ASHIRT servers. If you odn't know, please contact your administrator."

<img width="573" alt="Screen Shot 2020-08-27 at 3 36 12 PM" src="https://user-images.githubusercontent.com/26415/91501685-0faf7580-e87b-11ea-94c5-0470188a0a97.png">

Reply:
I'm assuming this is the `[]`? This is present to denote whatever value is either current, or being guessed at. The problem in your example is that your string is the empty string. I just revised this so that we won't show the guess if either the guess is `nil`, or the guess is empty, which should solve this.